plugins/modules/azure_rm_openshiftmanagedcluster_kubeconfig_info.py

```diff
@@ -0,0 +1,235 @@
+            - Destination filepath of kubeconfig file
+        required: false
+        type: str
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+author:
+    - Maxim Babushkin (@maxbab)
+'''
+
+EXAMPLES = '''
+    - name: Obtain kubeconfig file of ARO cluster
+      azure_rm_openshiftmanagedcluster_kubeconfig_info:
+        name: myCluster
+        resource_group: myResourceGroup
+      register: kubeconf
+
+    - name: Print registered kubeconfig file
+      debug:
+        msg: "{{ kubeconf['kubeconfig'] }}"
+
+    - name: Fetch kubeconfig and save it as mycluster_kubeconfig filename
+      azure_rm_openshiftmanagedcluster_kubeconfig_info:
+        name: myCluster
+        resource_group: myResourceGroup
+        path: ./files/mycluster_kubeconfig
+
+    - name: Fetch kubeconfig and save it to specified directory (file will be named as kubeconfig by default)
+      azure_rm_openshiftmanagedcluster_kubeconfig_info:
+        name: myCluster
+        resource_group: myResourceGroup
+        path: ./files/
+'''
+
+RETURN = '''
+kubeconfig:
+    description:
+        - kubeconfig value
+    returned: always
+    type: str
+'''
+
+import base64
+import filecmp
+import json
+import os
+import tempfile
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBaseExt
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_rest import GenericRestClient
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # this is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+class AzureRMOpenShiftManagedClustersKubeconfigInfo(AzureRMModuleBaseExt):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str', required=True
+            ),
+            name=dict(
+                type='str', required=True
+            ),
+            path=dict(
+                type='str', required=False
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.path = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.url = None
+        self.status_code = [200]
+
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = '2021-09-01-preview'
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        self.mgmt_client = None
+        super(AzureRMOpenShiftManagedClustersKubeconfigInfo, self).__init__(self.module_arg_spec, supports_check_mode=True, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
```

Comment:
All auth has support trunck2, Please add this parameters!
```suggestion
        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
                                                                                is_track2=True,
```

Reply:
Done.